dissect/target/filesystems/zip.py

```diff
@@ -0,0 +1,136 @@
+import io
+import logging
+import zipfile
+
+from dissect.util.stream import BufferedStream
+
+from dissect.target.exceptions import FileNotFoundError, FilesystemError
+from dissect.target.filesystem import Filesystem, VirtualFile, VirtualFilesystem
+from dissect.target.helpers import fsutil
+
+log = logging.getLogger(__name__)
+
+
+class ZipFilesystem(Filesystem):
+    """This Zip filesystem implementation is WIP and
+    has only been tested for opening java jar archives.
+    Its structure has been copied from the tar filesystem
+    implementation.
+    """
+
+    __fstype__ = "zip"
+
+    def __init__(self, fh, base=None, case_sensitive=True, zipinfo=None, *args, **kwargs):
+        fh.seek(0)
+
+        self.zip = zipfile.ZipFile(fh, mode="r")
+        self.base = base.strip("/") if base else ""
+
+        self._fs = VirtualFilesystem(case_sensitive=case_sensitive)
+
+        for member in self.zip.infolist():
+
+            if member.is_dir():
+                continue
+
+            mname = member.filename.strip("/")
+            if not mname.startswith(self.base):
+                continue
+
+            rel_name = mname[len(self.base) :]
+
+            # NOTE: Normally we would check here if the member is a symlink or not,
+            # however Python does not have symlink support in the zipfile module.
+            # see https://github.com/python/cpython/issues/82102
+
+            file_entry = ZipFilesystemEntry(self, rel_name, member)
+            self._fs.map_file_entry(rel_name, file_entry)
+
+        super().__init__(case_sensitive=case_sensitive, *args, **kwargs)
+
+    @staticmethod
+    def detect(fh):
+        """Detect a zip file on a given file-like object."""
+        offset = fh.tell()
+        try:
+            fh.seek(0)
+            zipfile.ZipFile(fileobj=fh, mode="r")
+            return True
```

Comment:
```suggestion
            return zipfile.is_zipfile(fh)
```
Should be a bit more efficient.